src/__tests__/components.test.js

```diff
@@ -0,0 +1,180 @@
+  code: `
+import { FormattedMessage as T } from 'react-intl'
+
+console.log(<T defaultMessage="hello" />);
+`,
+}
+
+const nestedJSXTest = {
+  title: 'with FormattedMessage nested in other JSX',
+  code: `
+import { FormattedMessage } from 'react-intl'
+
+console.log(
+  <div>
+    <FormattedMessage defaultMessage="hello" />
+  </div>
+);
+`,
+}
+
+const throwWhenNotAnalyzableTest = {
+  title: 'throws if defaultMessage isn’t analyzable',
+  code: `
+import { FormattedMessage } from 'react-intl'
+
+const getMsg = () => 'hello';
+
+console.log(<FormattedMessage defaultMessage={getMsg()} />);
+`,
+  error: /\[React Intl Auto\] Messages must be statically evaluate-able for extraction/,
+  snapshot: false,
+}
+
+const notTransformIfNotImportedTest = {
+  title: 'does nothing if components not imported from react-intl',
+  code: `
+import any from 'any-module'
+
+console.log(<FormattedMessage defaultMessage={getMsg()} />);
+`,
+}
+
+const notTransformIfSpreadAttributeTest = {
+  title: 'does nothing if component props are spread',
+  code: `
+import { FormattedMessage } from 'react-intl'
+
+const props = { defaultMessage: 'hello' };
+
+console.log(<FormattedMessage {...props} />);
+`,
+}
+
+const tests = [
+  defaultTest,
+  multiUseTest,
+  withValueInMessageTest,
+  withVariableMessageTest,
+  importAsTest,
+  nestedJSXTest,
+  throwWhenNotAnalyzableTest,
+  notTransformIfNotImportedTest,
+  notTransformIfSpreadAttributeTest,
+]
+
+cases([
+  { title: 'default', tests },
+  {
+    title: 'removePrefix = "src"',
+    tests: [defaultTest],
+    pluginOptions: { removePrefix: 'src' },
+  },
+  {
+    title: 'removePrefix = "src/" -- with slash',
+    tests: [defaultTest],
+    pluginOptions: { removePrefix: 'src/' },
+  },
+  {
+    title: 'filebase = true',
+    tests: [defaultTest],
+    pluginOptions: { filebase: true },
+  },
+  {
+    title: 'includeExportName = true',
+    tests: [defaultTest],
+    pluginOptions: { includeExportName: true },
+  },
+  {
+    title: 'includeExportName = all',
+    tests: [defaultTest],
+    pluginOptions: { includeExportName: 'all' },
+  },
+  {
+    title: 'removePrefix = true, includeExportName = true',
+    tests: [defaultTest],
+    pluginOptions: { removePrefix: true, includeExportName: true },
+  },
+  {
+    title: 'removePrefix = false',
+    tests: [defaultTest],
+    pluginOptions: { removePrefix: false },
+  },
+  {
+    title: 'removePrefix = true, includeExportName = all',
+    tests: [defaultTest],
+    pluginOptions: { removePrefix: true, includeExportName: 'all' },
+  },
+  {
+    title: 'extractComments = false',
+    tests: [defaultTest],
+    pluginOptions: { extractComments: false },
+  },
+])
+
+function cases(
+  testCases: {
+    title: string,
+    tests: $ReadOnlyArray<{ title: string, code: string }>,
+  }[]
+) {
+  const defaultOpts = {
+    title: '',
+    plugin,
+    snapshot: true,
+    babelOptions: { filename, babelrc: true },
```

Comment:
use ` babelOptions: { filename, parserOpts: { plugins: ['jsx'] } },`
remove `babelrc: true`

Reply:
Oh cool... I understand the dependency removal now too thanks. What does it use to parse jsx under the hood?

Reply:
See https://github.com/babel/babel/blob/master/packages/babylon/README.md